substrate/frame/identity/src/migration.rs

```diff
@@ -116,11 +156,560 @@ pub mod v1 {
 		fn post_upgrade(state: Vec<u8>) -> Result<(), TryRuntimeError> {
 			let identities_to_migrate: u64 = Decode::decode(&mut &state[..])
 				.expect("failed to decode the state from pre-upgrade.");
-			let identities = IdentityOf::<T>::iter().count() as u64;
+			let identities = types_v1::IdentityOf::<T>::iter().count() as u64;
 			log::info!("post-upgrade expects '{}' identities to have been migrated.", identities);
 			ensure!(identities_to_migrate == identities, "must migrate all identities.");
 			log::info!(target: TARGET, "migrated all identities.");
 			Ok(())
 		}
 	}
 }
+
+pub mod v2 {
+	use super::*;
+	use frame_support::{
+		migrations::{MigrationId, SteppedMigration, SteppedMigrationError},
+		weights::WeightMeter,
+	};
+
+	type HashedKey = BoundedVec<u8, ConstU32<256>>;
```

Comment:
This type can be optimistic.

The `HashedKey` is used for the storages `IdentityOf` which is `Twox64Concat` of `AccountId` and `PendingUsernames` which is `Blake2_128Concat` of `Username` which is of length `MaxUsernameLength`.

So if I read the code correctly then if there is a username of length more than 256-16 (16 for blake 2 128) then the `unwrap` in the migration will fail no?

I think `HashedKey` is fine. but we should no `unwrap` and instead truncate.

Reply:
Initially I just set a high limit to not worry about it in testing, but in any reasonable scenario it seems it should be ok to shrink to 32 for general use within translations or even 16 for the particular use here. Also yes, the unwraps need to go away. WDYT?

Reply:
I think any number more than 32 is good to me. Just we should remember to truncate after the `translate_next`

Reply:
Fixed in https://github.com/paritytech/polkadot-sdk/pull/5554/commits/c3d22fe6911b685c5143ea8fc974449f4aadb677